src/main/java/Duke.java

```diff
@@ -1,10 +1,208 @@
+        tasksList.add(new Event(eventDescription, eventTime));
+    }
+
+
+    private static void addDeadline(String input) throws DukeInvalidArgumentException {
+        String deadlineInput;
+        try {
+            deadlineInput = input.trim().split("deadline")[1];
+        } catch (IndexOutOfBoundsException e) {
+            printDividerLine();
+            System.out.println("☹ OOPS!!! The description of a deadline cannot be empty.");
+            printDividerLine();
+            throw new DukeInvalidArgumentException();
+        }
+        String deadlineDescription, deadlineTime;
+        try {
+            deadlineDescription = deadlineInput.split("/by")[0].trim();
+            deadlineTime = deadlineInput.split("/by")[1].trim();
+        } catch (IndexOutOfBoundsException e) {
+            printDividerLine();
+            System.out.println("☹ I cannot help you unless you provide the details!");
+            printDividerLine();
+            throw new DukeInvalidArgumentException();
+        }
+        tasksList.add(new Deadline(deadlineDescription, deadlineTime));
+    }
+
+    public static void printTasksList() {
+        printDividerLine();
+        System.out.println("Here are the tasks in your list:");
+        int i = 1;
+        for (Task item: tasksList) {
+            System.out.println((i++) + ". " + item);
+        }
+        printDividerLine();
+    }
+
+    public static void deleteTask(int index) {
+        printDividerLine();
+        System.out.println("Got it! I've removed this task:");
+        System.out.println(tasksList.get(index - 1));
+        tasksList.remove(index - 1);
+        System.out.println("Now you have " + tasksList.size() + " tasks in the list.");
+        printDividerLine();
+    }
+
+    public static void markTaskAsDone(int index) throws DukeInvalidArgumentException {
+        try {
+            tasksList.get(index - 1).markAsDone();
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeInvalidArgumentException();
+        }
+        printDividerLine();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println("[X] " + tasksList.get(index - 1).getDescription());
+        printDividerLine();
+    }
+
+    public static String handleInput(String input) {
+        String[] wordsInInputString = input.split(" ");
+        if (input.equals("bye")) {
+            return "End";
+        }
+        if (input.equals("list")) {
+            printTasksList();
+        } else if (wordsInInputString[0].equals("done")) {
+            int index;
+            try {
+                index = Integer.parseInt(input.split("done")[1].trim());
+                markTaskAsDone(index);
+            } catch (DukeInvalidArgumentException e) {
+                printDividerLine();
+                System.out.println("☹ OOPS!!! No such item in the list :-(");
+                printDividerLine();
+            }
+        } else if (input.startsWith("delete")){
```

Comment:
Including the logic inside this function would cause the function to be quite long (especially if the list of commands grows). Would recommend extracting out into smaller functions that handle each type of command